src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -1154,3 +1154,71 @@ def restore_storage_account(client, vault_base_url, file_path):
         data = file_in.read()
         return client.restore_storage_account(vault_base_url, data)
 # endregion
+
+
+# region private_endpoint
+def show_private_endpoint_connection(client, resource_group_name, vault_name, connection_name,
+                                     connection_id=None):  # pylint: disable=unused-argument
+    """Show details of a private endpoint connection associated with a Key Vault."""
+    return client.get(resource_group_name=resource_group_name, vault_name=vault_name,
+                      private_endpoint_connection_name=connection_name)
+
+
+def delete_private_endpoint_connection(client, resource_group_name, vault_name, connection_name,
+                                       connection_id=None):  # pylint: disable=unused-argument
+    """ Delete the specified private endpoint connection associated with a Key Vault."""
+    return client.delete(resource_group_name=resource_group_name, vault_name=vault_name,
+                         private_endpoint_connection_name=connection_name)
+
+
+def approve_private_endpoint_connection(cmd, client, resource_group_name, vault_name, connection_name,
+                                        approval_description=None,
+                                        connection_id=None):  # pylint: disable=unused-argument
+    """Approve a private endpoint connection request for a Key Vault."""
+
+    PrivateEndpointConnection = cmd.get_models('PrivateEndpointConnection', resource_type=ResourceType.MGMT_KEYVAULT)
+    PrivateLinkServiceConnectionState = cmd.get_models('PrivateLinkServiceConnectionState',
+                                                       resource_type=ResourceType.MGMT_KEYVAULT)
+
+    private_endpoint_connection = client.get(resource_group_name=resource_group_name, vault_name=vault_name,
+                                             private_endpoint_connection_name=connection_name)
+
+    return client.put(resource_group_name=resource_group_name,
+                      vault_name=vault_name,
+                      private_endpoint_connection_name=connection_name,
+                      private_endpoint=private_endpoint_connection.private_endpoint,
+                      properties=PrivateEndpointConnection(
+                          tags=private_endpoint_connection.tags,
+                          location=private_endpoint_connection.location,
+                          private_link_service_connection_state=PrivateLinkServiceConnectionState(
+                              status='Approved',
+                              description=approval_description
+                          )
+                      ))
+
+
+def reject_private_endpoint_connection(cmd, client, resource_group_name, vault_name, connection_name,
```

Comment:
We'd better to extract the common code and call it with a switch.

Reply:
Good idea.